packages/client-core/src/networking/NetworkInstanceProvisioning.tsx

```diff
@@ -236,14 +240,30 @@ export const PartyInstanceProvisioning = () => {
   return null
 }
 
+export const WorldNetworkPeer = (props: { userID: UserId; name: string }) => {
```

Comment:
Reactively grab username here instead of passing down as prop, and only dispatch the notification in the effect when the userid has a username. Can add a boolean state to see if the user has not already been announced, and otherwise dispatch a notification that they changed their name.